mne/utils/check.py

```diff
@@ -17,10 +17,32 @@
 
 import numpy as np
 
-from ..fixes import _median_complex, _compare_version
+from ..fixes import _median_complex
 from ._logging import warn, logger, verbose
 
 
+def _compare_version(version_a, operator, version_b):
+    """Compare two version strings via a user-specified operator.
+
+    Parameters
+    ----------
+    version_a : str
+        First version string.
+    operator : '==' | '>' | '<' | '>=' | '<='
+        Operator to compare ``version_a`` and ``version_b`` in the form of
+        ``version_a operator version_b``.
+    version_b : str
+        Second version string.
+
+    Returns
+    -------
+    bool
+        The result of the version comparison.
+    """
+    from packaging.version import parse  # noqa:F401
+    return eval(f'parse("{version_a}") {operator} parse("{version_b}")')
```

Comment:
Please restore the `catch_warnings` wrapper I added in https://github.com/mne-tools/mne-python/pull/10199#discussion_r786996171 that you can also see in the [diff above](https://github.com/mne-tools/mne-python/pull/10223/files#diff-50629b846cfe5156a9681aa5a131f1511f177e420981d19db50b2e6c67816e3eL55-L56). We (and users) don't need to hear a warning if package versions are not PEP440 compatible when using this function.

Reply:
I don't think `packaging.version.parse` raises this warning. Can you give an example of a problematic version and/or comparison?

Reply:
```
>>> import warnings
>>> warnings.simplefilter('always', DeprecationWarning)
>>> import packaging.version
>>> packaging.version.parse('0.1.develop')
/home/larsoner/.local/lib/python3.10/site-packages/packaging/version.py:111: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
  warnings.warn(
<LegacyVersion('0.1.develop')>
```

Reply:
Are `DeprecationWarnings` not shown by default (that's why I didn't see it)? Thanks for the explanation, I'll revert the change.

Reply:
Apparently yes: https://docs.python.org/3/library/warnings.html#default-warning-filter